### PR TITLE
MH-12904, Paella player 5.3 update

### PR DESCRIPTION
**Pull request by rrolf**
I tried to upgrade Paella Player to 5.3.x what needed quite some changes as now ECMAscript 6 seems to be required.

### CODE REVIEW

File: modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.downloadsPlugin/mh_downloads.js [hunk 1]
```diff
@@ -18,119 +18,119 @@
...
+        var download = false;
+        if (track.tags != undefined && track.tags.tag != undefined
+            && track.mimetype.indexOf("video") >= 0
+            && track.url.indexOf("rtmp://") < 0) {
+          for (var j = 0; j < track.tags.tag.length; j++) {
+            if (track.tags.tag[j] === "engage-download") {
+            download = true;
+              break;
+            }
+                }
```
Comment (lkiesow) [inline modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.downloadsPlugin/mh_downloads.js]: indentation

File: modules/engage-paella-player/src/main/paella-opencast/ui/watch.html [hunk 2]
```diff
@@ -4,8 +4,8 @@
 		<meta http-equiv="Content-type" content="text/html; charset=utf-8;">
 		<meta http-equiv="X-UA-Compatible" content="IE=edge">
 		<title>Opencast Paella Player</title>
-		<script type="text/javascript" src="javascript/swfobject.js"></script>
-		<script type="text/javascript" src="javascript/traceur-compiler.js"></script>
+		<script type="text/javascript" src="javascript/traceur-runtime.min.js"></script>
+    <script type="text/javascript" src="javascript/swfobject.js"></script>
```
Comment (lkiesow) [inline modules/engage-paella-player/src/main/paella-opencast/ui/watch.html]: indentation

File: modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.transcriptionTabBarPlugin/mh_transcription.js [hunk 3]
```diff
@@ -18,298 +18,308 @@
...
+
+            if (!thisClass.foundAlready) {
+              thisClass.foundAlready = true;
+            }
+            thisClass.lastHit = value;
+          }
+          else {
+            paella.debug.log("No Revelance");
+            if (!thisClass.foundAlready){
+              //setNoSegmentDataAvailable();
```
Comment (lkiesow) [inline modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.transcriptionTabBarPlugin/mh_transcription.js]: empty condition

**Review by lkiesow — COMMENTED** [review]
A few things I noticed while skimming through the code

**Comment by staubesv** [comment]
Hi @rrolf 

As this has been lying around for quite a while, I have taken the review. I already started to write comments about the wrong formatting of the JSONs.... just to find out that they seem to come from the Paella repository.

Can you please provide me an overview of what files come from where?

It seems that the JSONs come from the official Paella Player standalone repository (https://github.com/polimediaupv/paella) whereas the Javascript files come from the Opencast Paella Player repository (https://github.com/polimediaupv/paella-opencast/) and have been ported to ECMA 6.

Is this correct?

What will happen to https://github.com/polimediaupv/paella-opencast/? Will that repository be removed?

**Comment by miesgre** [comment]
Hi @staubesv, 

Let me explain... The opencast paella module (https://github.com/opencast/opencast/tree/develop/modules/engage-paella-player) is a rewrite of the  opencast paella player repository (https://github.com/polimediaupv/paella-opencast/). 
As paella bundle is now in opencast tree, we does not plan to maintain `polimediaupv/paella-opencast` repository anymore (https://github.com/polimediaupv/paella-opencast/issues/39)


All development of paella is done in paella official repository (https://github.com/polimediaupv/paella). In opencast, there is no paella source as paella player is downloaded using npm (https://github.com/opencast/opencast/blob/develop/modules/engage-paella-player/package.json#L8)

So, which JSON files have wrong format? In paella opencast module there are only a `package.json` file and the `localization JSON files`

**Comment by staubesv** [comment]
Hi @miesgre 

Thanks for the explanations!

I mainly meant the config.json that does not have a consistent formatting. Sometimes, for example, you find things like '"object":{' vs '"object": {' or '"object": true' vs '"object":true'. Sometimes, objects are extended across multiple lines, most of the time they are not.

This is not wrong, but it is not following a consistent formatting rule. Later I found that config.json has the same "issues" in the Paella repository, too.

File: modules/engage-paella-player/src/main/paella-opencast/ui/watch.html [hunk 4]
```diff
@@ -4,8 +4,8 @@
 		<meta http-equiv="Content-type" content="text/html; charset=utf-8;">
 		<meta http-equiv="X-UA-Compatible" content="IE=edge">
 		<title>Opencast Paella Player</title>
-		<script type="text/javascript" src="javascript/swfobject.js"></script>
-		<script type="text/javascript" src="javascript/traceur-compiler.js"></script>
+		<script type="text/javascript" src="javascript/traceur-runtime.min.js"></script>
+                <script type="text/javascript" src="javascript/swfobject.js"></script>
```
Comment (staubesv) [inline modules/engage-paella-player/src/main/paella-opencast/ui/watch.html]: Indentation?

**Review by staubesv — APPROVED** [review]

**Comment by staubesv** [comment]
Hi @miesgre & @rrolf 

I did some basic tests which worked -> merged.

**Pull request merged**